GillianCore/utils/option_utils.ml

```diff
@@ -11,3 +11,8 @@ let somes_and_eq ox y =
 let or_else f = function
   | Some x -> x
   | None -> f ()
+
+let coalesce a b =
+  match (a, b) with
+  | Some x, _ | _, Some x -> Some x
+  | _ -> None
```

Comment:
```suggestion
let coalesce a b =
  match a with
  | Some _ -> a
  | _ -> b
```
equivalent solution, but lighter.
First, matching on b is not necessary, so that's an operation you can avoid
Second, in the case when you return a, you don't have to rebuild a `Some x` as the `a` allocation may simply be reused

Reply:
acknowledging that was my own code 😆

Reply:
Changed in 04099f5707f52ded44d805023b721f07f97650a2